weekly_data_report.R

```diff
@@ -0,0 +1,131 @@
+
+# INTAKE_ID
+# SCREENING_DECISION = SCREENED OUT or SCREENED IN
+# ALLEGATION_ADDRESS, formatted in a string without punctuation, it includes city, state, and zip code
+
+message("\nNeighborhood results with < 5 instances have been censored for privacy purposes\n")
+
+suppressPackageStartupMessages(library(readr))
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(sf))
+suppressPackageStartupMessages(library(knitr))
+
+d <- read_csv(args$file_name,
+              col_types = cols(INTAKE_ID = col_character(),
+                               SCREENING_DECISION = col_character(),
+                               DECISION_DATE = col_character(),
+                               PERSON_ID = col_character(),
+                               RACE = col_character(),
+                               ADDRESS_START = col_date(),
+                               MANDATED_REPORTER = col_character(),
+                               #  REPORTER_PERSON_ID = col_character(),
+                               address_type = col_character(),
+                               address = col_character(),
+                               bad_address = col_logical(),
+                               PO = col_logical(),
+                               lat = col_double(),
+                               lon = col_double(),
+                               score = col_double(),
+                               precision = col_character(),
+                               precise_geocode = col_logical(),
+                               fips_tract_id = col_character(),
+                               fraction_assisted_income = col_double(),
+                               fraction_high_school_edu = col_double(),
+                               median_income = col_double(),
+                               fraction_no_health_ins = col_double(),
+                               fraction_poverty = col_double(),
+                               fraction_vacant_housing = col_double(),
+                               dep_index = col_double()
+              ))
+
+###
+# d_test <- read.csv('test/simulated_jfs_data_geocoded.csv')
+# d <- d_test
+###
+d <- dplyr::mutate(d, DECISION_DATE = dht::check_dates(DECISION_DATE))
+
+# rmarkdown::render(input = '/app/aggregate_data_report.rmd',
+#                   params = list(d = d),
+#                   envir = new.env())
+
+
+tract_to_neighborhood <- readRDS('/app/tract_to_neighborhood.rds')
+neighborhood_shp <- readRDS('/app/ham_neighborhoods_dep_index_shp.rds')
+
+# Overall Summary
+date_min <- min(d$DECISION_DATE)
+date_max <- max(d$DECISION_DATE)
+
+# consider 'SCREENED IN AR' same as 'SCREENED IN'
+for (i in 1:nrow(d)) {
+  if (d$SCREENING_DECISION[i] == 'SCREENED IN AR') {
+    d$SCREENING_DECISION[i] <- 'SCREENED IN'
+  }
+}
+
+d_missing_alleg_add <- d %>%
+  filter(address_type == 'ALLEGATION_ADDRESS',
+         is.na(address)) %>%
+  select(PERSON_ID:address_type)
+
+d_fill_in_address <- d %>%
```

Comment:
Can you summarize this logic in the README.md, just so we have a clear record of it.  e.g., 

- set `SCREENED IN AR` to `SCREENED IN`
- use `CHILD_ADDRESS` if `ALLEGATION_ADDRESS` is missing